PowerUp/app/src/main/java/powerup/systers/com/pregame_setup/PregameSetupActivity.java

```diff
@@ -0,0 +1,212 @@
+package powerup.systers.com.pregame_setup;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AbsListView;
+import android.widget.AdapterView;
+import android.widget.BaseAdapter;
+import android.widget.GridView;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import powerup.systers.com.R;
+import powerup.systers.com.datamodel.SessionHistory;
+import powerup.systers.com.powerup.PowerUpUtils;
+
+import static powerup.systers.com.powerup.PowerUpUtils.MAX_ELEMENTS_PER_SCREEN;
+
+public class PregameSetupActivity extends AppCompatActivity {
+
+    private GridView gridView;
+    private List<List<Integer>> npcViewList = new ArrayList<>();
+    private List<List<Integer>> npcFullViewList = new ArrayList<>();
+    private GridAdapter gridViewAdapter;
+    private ImageView imageViewNpc, backButton, leftArrow, rightArrow, storeTick;
+    private TextView npcText;
+    private int screenWidth, screenHeight, currentPage, previousClickedPosition = 0, clickedPosition = 0;
+    private boolean firstClick = true;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```

Comment:
I need suggestion to rectify this.

Reply:
I don't think this is an issue in this method since it's declaring some variables and some setups. However, if you want to take more care of the simplicity and the quality of the code, you can follow a certain method template. For example, `private void setupListeners()` method will have all addition of click listeners.